ports/atmel-samd/common-hal/busio/SPI.c

```diff
@@ -57,6 +57,10 @@ void common_hal_busio_spi_construct(busio_spi_obj_t *self,
     uint8_t miso_pad = 0;
     uint8_t dopo = 255;
 
+    if (half_duplex == true) {
```

Comment:
```suggestion
    if (half_duplex) {
```
This is more canonical C.